_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title:  "Announcing Data Prepper 2.3.0"
+authors:
+- kkondaka
+- dlv
+date: 2023-06-06 10:00:00 -0700
+categories:
+  - releases
+meta_keywords: Data Prepper 2.3.0, ingestion, observability, log pipelines
+meta_description: Data Prepper 2.3.0 with enhancements to expressions, event tagging, enhancements to add_entries processor, s3 sink and tail sampling processor
+---
+
+Data Prepper 2.3.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper)!
+This release introduces a number of changes that help with Data Prepper’s ability to process complex expressions with functions, arithmetic operations, and string operations.
+
+## Enhancements to Data Prepper expressions
+
+Data Prepper 2.3 supports using functions in expressions. A list of supported functions is available in the [Expression syntax](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) documentation.
+
+Data Prepper 2.3 supports three types of expressions:
+
+* Conditional expressions
+    * Conditional expressions evaluate to a result of Boolean type, and the expressions now can have functions in them. For example, `length(/message) > 20` would evaluate to true if the length of the message field in the event is greater than 20. Otherwise it evaluates to false. 
+* Arithmetic expressions
+    * Arithmetic expressions evaluate to a result of integer or float type. The expressions can have simple arithmetic operators like `+,-,*,/` with functions or JSON pointers or literals as operands. For example, the following expression adds the length of message field of string type in the event with the value of event metadata with the key `integerField` and subtracts 1 from it.
+
+ ```length(/message) + getMetadata("integerField") - 1```
+
+* String expressions
+    * String expressions evaluate to a result of string type. The string concatenation operator is supported in addition to using functions or JSON pointers or literals as operands. For example, the following expression adds the message1 field of type string in the event with message2 field of type string in the event and appends "suffix" to it.
+
+```/message1 + /message2 + "suffix"```
+
+## Event tagging
+
+Data Prepper 2.3 supports tagging events while using the `grok` processor. Events can be tagged optionally using the following configuration
+
+```
+processor:
+   - grok: 
+        match:
+          message: <pattern>
+        tags_on_match_failure: ["grok_match_fail", "another_tag"]
+```
+
+The presence of tags can be checked in conditional expressions in different processors or routing using `hasTags()` function to do conditional ingestion. For example, a conditional expression checking for the presence of tag `grok_match_fail` would be `hasTags("grok_match_fail")`
+
+## Enhancements to `add_entries` processor
+
```

Comment:
These headings should ideally be separated by text so that they are not "stacked."